src/grasp_data.cpp

```diff
@@ -112,9 +109,32 @@ bool GraspData::loadGraspData(const ros::NodeHandle& nh, const std::string& end_
   rosparam_shortcuts::get(parent_name, child_nh, "grasp_posture", grasp_posture);
   rosparam_shortcuts::get(parent_name, child_nh, "grasp_pose_to_eef_transform", grasp_pose_to_eef_pose_);
   rosparam_shortcuts::get(parent_name, child_nh, "grasp_padding_on_approach", grasp_padding_on_approach_);
-  rosparam_shortcuts::get(parent_name, child_nh, "max_finger_width", max_finger_width_);
-  rosparam_shortcuts::get(parent_name, child_nh, "min_finger_width", min_finger_width_);
 
+  // Find out if the end effector uses suction or fingers (NOTE: must be one of 'finger' or 'suction')
+  std::string end_effector_type_str;
+  child_nh.param<std::string>("end_effector_type", end_effector_type_str, "finger");
+
+  if (end_effector_type_str == "finger")
+    end_effector_type_ = FINGER;
+  else if (end_effector_type_str == "suction")
+    end_effector_type_ = SUCTION;
+  else
+    ROS_ASSERT_MSG(false, "Unrecognized end effector type: %s", end_effector_type_str.c_str());
+
+  if (end_effector_type_ == FINGER)
+  {
+    rosparam_shortcuts::get(parent_name, child_nh, "finger_to_palm_depth", grasp_max_depth_);
+    rosparam_shortcuts::get(parent_name, child_nh, "gripper_finger_width", gripper_finger_width_);
+    rosparam_shortcuts::get(parent_name, child_nh, "max_grasp_width", max_grasp_width_);
+    rosparam_shortcuts::get(parent_name, child_nh, "max_finger_width", max_finger_width_);
+    rosparam_shortcuts::get(parent_name, child_nh, "min_finger_width", min_finger_width_);
+  }
+  else if (end_effector_type_ == SUCTION)
+  {
+    rosparam_shortcuts::get(parent_name, child_nh, "active_suction_range_x", active_suction_range_x_);
+    rosparam_shortcuts::get(parent_name, child_nh, "active_suction_range_y", active_suction_range_y_);
+    rosparam_shortcuts::get(parent_name, child_nh, "suction_cup_stroke", grasp_max_depth_);
```

Comment:
These, and previous ``get()`` functions, should check for missing params and exit if error. This is the normal pattern, not sure why it didn't get implemented here...

int error = 0;
error += rosparam_shortcuts::get(...)
if error
    exit

Reply:
9ce13f9d0f58386c23fdf300a43d510052404732